search/retrieval.cpp

```diff
@@ -74,29 +119,44 @@ unique_ptr<coding::CompressedBitVector> RetrieveAddressFeaturesImpl(
   // bit vectors are sorted in the search index.
   vector<uint64_t> features;
   uint32_t counter = 0;
-  auto collector = [&](TValue const & value)
+  auto const collector = [&](TValue const & value)
   {
     if ((++counter & 0xFF) == 0)
       BailIfCancelled(cancellable);
     features.push_back(value.m_featureId);
   };
 
-  MatchFeaturesInTrie(params, *trieRoot, emptyFilter, collector);
+  MatchFeaturesInTrie(params, *trieRoot, filter, collector);
+  // Match all edited/created features separately.
+  auto const created = editor.GetFeaturesByStatus(id, Editor::FeatureStatus::Created);
```

Comment:
What the point to get vector and then create a copy of it at the |modified| end? Why not to move loop body to a named lambda and then iterate on modified and created via for_each? The code will be short too, but without redundant copying.